Mathlib/Order/OmegaCompletePartialOrder.lean

```diff
@@ -115,6 +115,25 @@ lemma isChain_range : IsChain (· ≤ ·) (Set.range c) := Monotone.isChain_rang
 
 lemma directed : Directed (· ≤ ·) c := directedOn_range.2 c.isChain_range.directedOn
 
+/-- An example of a `Chain` constructed from an ordered pair. -/
+def ordered_pair_to_chain (a : α) (b : α) (hab : a ≤ b) : Chain α :=
+  ⟨fun n => match n with
+    | 0 => a
+    | _ => b, fun _ _ _ => by aesop⟩
+
+lemma ordered_pair_to_chain_range (a : α) (b : α) (hab : a ≤ b) :
+    Set.range (ordered_pair_to_chain a b hab) = {a , b} :=  by
```

Comment:
```suggestion
@[simp] lemma range_pair (a b : α) (hab : a ≤ b) :
    Set.range (pair a b hab) = {a , b} := by
  ext; simp [pair, Nat.and_forall_succ]
```

Reply:
Sorry, I can't see what to do with this:
```
@[simp] lemma range_pair (a b : α) (hab : a ≤ b) :
    Set.range (pair a b hab) = {a , b} := by
  ext m; simp [pair, Nat.and_forall_succ]
  constructor
  · intro h
    sorry
  · intro h
    cases' h with ha hb
    · use 0; rw [ha]; rfl
    · use 1; rw [hb]; rfl
```
At the `sorry` I have:
```
m : α
h : ∃ y,
  {
        toFun := fun n ↦
          match n with
          | 0 => a
          | x => b,
        monotone' := ⋯ }
      y =
    m
⊢ m = a ∨ m = b
```

Reply:
Sorry, I meant
```
@[simp] lemma range_pair (a b : α) (hab : a ≤ b) :
    Set.range (pair a b hab) = {a , b} := by
  ext m
  simp? [pair]
  rw [← Nat.or_exists_succ]
  simp
```
Probably golfable further

Reply:
This works:
```
lemma pair_zero (a b : α) (hab : a ≤ b) : pair a b hab 0 = a := rfl

lemma pair_n_plus_one (a b : α) (hab : a ≤ b) (n : ℕ) : pair a b hab (n + 1) = b := rfl

@[simp] lemma range_pair (a b : α) (hab : a ≤ b) : Set.range (pair a b hab) = {a, b} := by
  ext;
  exact Nat.or_exists_succ.symm.trans (by simp_rw [pair_zero, pair_n_plus_one]; aesop)
```